src/lib.rs

```diff
@@ -101,31 +104,31 @@ fn split_hash(hash: &str) -> BcryptResult<HashParts> {
 
 /// Generates a password hash using the cost given.
 /// The salt is generated randomly using the OS randomness
-pub fn hash(password: &str, cost: u32) -> BcryptResult<String> {
+pub fn hash(password: impl AsRef<[u8]>, cost: u32) -> BcryptResult<String> {
     let salt = {
         let mut s = [0u8; 16];
         let mut rng = OsRng::new()?;
         rng.fill_bytes(&mut s);
         s
     };
-    let hash_parts = _hash_password(password, cost, &salt)?;
+    let hash_parts = _hash_password(password.as_ref(), cost, &salt)?;
 
     Ok(hash_parts.format())
 }
 
 /// Verify that a password is equivalent to the hash provided
-pub fn verify(password: &str, hash: &str) -> BcryptResult<bool> {
+pub fn verify(password: impl AsRef<[u8]>, hash: &str) -> BcryptResult<bool> {
```

Comment:
can we go with non `impl` version of `AsRef` to keep the compatibility with older versions of Rust?

Reply:
Done